core/sts-3.3.md

```diff
@@ -0,0 +1,94 @@
+---
+title: IRCv3.3 Strict Transport Security
+layout: spec
+copyrights:
+  -
+    name: "Attila Molnar"
+    period: "2015"
+    email: "attilamolnar@hush.com"
+---
+
+## Description
+
+Strict Transport Security (STS) is a mechanism which allows servers to
+communicate that complying clients should only connect to them over a secure
+connection.
+
+The policy is communicated to the client via the `sts` capability and should
+be processed by the client at capability negotiation time.
+
+The value of the capability specifies the duration during which the client
+MUST only connect to the server securely (using TLS, aka. SSL), and the port
+number to use for it.
+
+## Details
+
+The capability value, which MUST always be present, is a comma (`,`) separated
+list of tokens. Each token is a key which may have a value attached. If there
+is a value attached, the value is separated from the key by a `=`. That is,
+`<key>[=<value>][<key2>[=<value2>][<keyN>[=<valueN>]]]`. Keys specified in
+this document MUST only occur at most once.
+
+See [capability negotiation 3.2](capability-negotiation-3.2.html) for more
+information about capabilities with values.
+
+When the client sees the `sts` capability advertised over a non-secure
+connection, it MUST first establish a secure connection and confirm that the
+`sts` capability is still present.
```

Comment:
If there is an MITM attack directing you to connect to a specific port, that to me is a red flag that you someone is trying to direct you to e.g. a compromised port. As such, no sts cap advertised on insecure protocols should be trusted, beyond the unavoidable necessity of actually trying to connect to that port.

This is, as hinted in the PR, where HSTS differs from IRCSTS. See http://tools.ietf.org/html/rfc6797#section-8.1
> If an HTTP response is received over insecure transport, the UA MUST ignore any present STS header field(s).

And also see http://tools.ietf.org/html/rfc6797#section-14.5 about the risks of obeying compromised directives.

Basically, there's no guarantee that the secure port you've been connected to is legitimate, so using it shouldn't be the default. This is probably a case where a client should present a warning to the user.